GillianCore/debugging/utils/exec_map.ml

```diff
@@ -1,38 +1,62 @@
 module L = Logging
```

Comment:
How about some module-level documentation about what an exec map is, like what it's for, what it looks like.
The annotations on the type lack context